spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/discovery/DiscoveryClientRouteDefinitionLocator.java

```diff
@@ -93,8 +95,8 @@ public DiscoveryClientRouteDefinitionLocator(DiscoveryClient discoveryClient,
 			};
 		}
 
-		return Flux.fromIterable(discoveryClient.getServices())
-				.map(discoveryClient::getInstances)
+		return Mono.fromSupplier(discoveryClient::getServices)
```

Comment:
1) it would be better to use `Flux.defer` + `Flux.fromIterable`
2) try to keep `subscribeOn` as close to the blocking call as possible (in this case, right after `defer`

Reply:
`.map(discoveryClient::getInstances)` is also a blocking call, shouldn't `subscribeOn` after `map`?

Reply:
Was not sure if `.map(discoveryClient::getInstances)` should also be above `subscribeOn` as it also blocks. Changed the code accordingly.

Reply:
`subscribeOn` will move the computation to the elastic scheduler, and `fromIterable` will publish everything on the same thread, which means that `.map` will also be executed on the elastic thread

Reply:
@bsideup Can I do sth. else to improve the PR?